src/ToricVarieties/NormalToricVarieties/constructors.jl

```diff
@@ -0,0 +1,243 @@
+######################
+# 1: The Julia type for ToricVarieties
+######################
+abstract type AbstractNormalToricVariety end
+
+struct NormalToricVariety <: AbstractNormalToricVariety
+           polymakeNTV::Polymake.BigObject
+end
+export NormalToricVariety
+
+struct AffineNormalToricVariety <: AbstractNormalToricVariety
+           polymakeNTV::Polymake.BigObject
+end
+export AffineNormalToricVariety
+
+
+function pm_ntv(v::AbstractNormalToricVariety)
+    return v.polymakeNTV
+end
+
+######################
+# 2: Generic constructors
+######################
+
+
+@doc Markdown.doc"""
+    AffineNormalToricVariety(C::Cone)
+
+Construct the affine normal toric variety $U_{C}$ corresponding to a polyhedral
+cone `C`.
+
+# Examples
+Set `C` to be the positive orthant in two dimensions.
+```jldoctest
+julia> C = Oscar.positive_hull([1 0; 0 1])
+A polyhedral cone in ambient dimension 2
+
+julia> antv = AffineNormalToricVariety(C)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+```
+"""
+function AffineNormalToricVariety(C::Cone)
+    pmc = Oscar.pm_cone(C)
+    fan = Polymake.fan.check_fan_objects(pmc)
+    pmntv = Polymake.fulton.NormalToricVariety(fan)
+    return AffineNormalToricVariety(pmntv)
+end
+
+
+@doc Markdown.doc"""
+    NormalToricVariety(C::Cone)
+Construct the (affine) normal toric variety $X_{\Sigma}$ corresponding to a
+polyhedral fan $\Sigma = C$ consisting only of the cone `C`.
+# Examples
+Set `C` to be the positive orthant in two dimensions.
+```jldoctest
+julia> C = Oscar.positive_hull([1 0; 0 1])
+A polyhedral cone in ambient dimension 2
+julia> ntv = NormalToricVariety(C)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+```
+"""
+function NormalToricVariety(C::Cone)
+    return AffineNormalToricVariety(C)
+end
+
+
+@doc Markdown.doc"""
+    NormalToricVariety(PF::PolyhedralFan)
+
+Construct the normal toric variety $X_{PF}$ corresponding to a polyhedral fan `PF`.
+
+# Examples
+Take `PF` to be the normal fan of the square.
+```jldoctest
+julia> square = Oscar.cube(2)
+A polyhedron in ambient dimension 2
+
+julia> nf = Oscar.normal_fan(square)
+A polyhedral fan in ambient dimension 2
+
+julia> ntv = NormalToricVariety(nf)
+A normal toric variety corresponding to a polyhedral fan in ambient dimension 2
+```
+"""    
+function NormalToricVariety(PF::PolyhedralFan)
+    fan = Oscar.pm_fan(PF)
+    pmntv = Polymake.fulton.NormalToricVariety(fan)
+    if fan.N_MAXIMAL_CONES == 1
+        return AffineNormalToricVariety( pmntv )
```

Comment:
Same as above.

Reply:
> Same as above.

As I said in the meeting yesterday, In general, I am not fond at all of such an "intelligent" design.  The reason is that this is very likely to lead to errors downstream, which are difficult to track down for less experienced users.

In this particular case this looks like it could be justified. But, if you want to keep it: please document this behavior super-precisely!

However, you could ask why toric varieties are so special.  A more systematic solution would be that any polyhedral fan with a single maximal cone will be treated as a cone. And then any function which rightfully expects a fan fails (or you write all downstream functions twice).

Reply:
PS: None of this should stop the merge.  But prepare for a (minor) refactoring.